tools/lint/check_spdx_header_test.py

```
@@ -0,0 +1,213 @@
+    def test_a_raising_enumeration_returns_no_verdict(self, monkeypatch):
+        """MUTATION TARGET. Drop the try/except in ``main`` and this returns 1.
+
+        ``tracked_paths`` raises rather than returning ``[]``, so the failure
+        arrives as an exception. An unhandled one exits **1**, and 1 already
+        means "headers are missing" -- a claim about the tree. Enumeration
+        failing is the opposite claim: the tree could not be read at all.
+        Collapsing them would undo the separation this check exists to make.
+        """
+
+        def _boom() -> list:
+            raise OSError("git ls-files unavailable")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)